crd/fake/cro.go

```diff
@@ -0,0 +1,30 @@
+package fake
+
+import (
+	"encoding/json"
+	"fmt"
+
+	apismetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func NewCRO(ID string) []byte {
+	newCustomObject := &CustomObject{
+		TypeMeta: apismetav1.TypeMeta{
+			APIVersion: Group + `/` + VersionV1,
+			Kind:       Kind,
+		},
+		ObjectMeta: apismetav1.ObjectMeta{
+			Name: ID,
+		},
+		Spec: Spec{
+			ID: ID,
+		},
+	}
+
+	b, err := json.Marshal(newCustomObject)
+	if err != nil {
+		panic(fmt.Sprintf("%#v", err))
+	}
```

Comment:
This is a bit odd. We use a rest client to create/delete/update custom objects. With 1.8 I am not able to get the rest client encoder working to accept a custom object as `runtime.Object`. That is why I just use the json library to marshal it.